DGP_CVXPY_Code/CS_amp_optimize_BW.py

```diff
@@ -1,61 +1,59 @@
 import cvxpy as cp
 import numpy as np
+import matplotlib.pyplot as plt
 
-# Constants
-VTO = 0.5 # Threshold voltage
-KP = 50e-6   # Transconductance parameter (μnCox)
-Cox = 2.3e-3
-mu_n = KP/Cox #Mobility in m^2/Vs
+# Constants and assumptions for 1um technology
+L = 1e-6  # Channel length (m)
+CL = 1e-12  # Load capacitance (F)
 
-# Given parameters
-Rs = 10e3  # Source resistance in Ohms
-RD = 5e3  # Drain resistance in Ohms
-CL = 1e-12  # Load capacitance in Farads
+VTO = 0.5  # Threshold voltage in Volts
+KP = 50e-6  # Transconductance parameter in A/V^2
+Cox = 2.3e-3  # Oxide capacitance per unit area in F/m^2
+mu_n = KP / Cox  # Mobility in m^2/Vs
+RS = 10e3 # Source resistance resistance (Ω)
+VoV = 0.2
 
 
-# Create optimization variables
-L = cp.Variable(pos=True)  # Width of the transistor
-VOV = cp.Variable(pos=True) # Gate-source voltage
+# Define the optimization variable
+W = cp.Variable(pos=True)
 
-# Specification 
-ID = 500e-6
-AV = 4
+# Define expressions for our metrics
+gain = 4
+gm = mu_n * Cox * (W/L) * VoV
+Rout = gain/gm
+Cgs = 2/3 * Cox * W * L
+f3db = 1 / (2 * np.pi * (Cgs*RS + CL*Rout))
+Id = 0.5 * mu_n * Cox * (W/L) * VoV**2
+# Id = gm * VoV / 2
 
-W = AV * L  / (RD * mu_n * VOV * Cox)
-Rout = RD
-omega_3dB = 1 / (((2/3)*cp.square(L)*AV*Rs/(RD*mu_n*VOV)) + AV*CL*VOV/(2*ID))
+#Contraint on the minimum f3db
+f3db_min = 10e6

```

Comment:
Do we need f3db_min? How would you know which value to pick a priori?

Reply:
We can omit the f3db_min and just focus on the optimum or maximum achievable f3db. I believe that specifying it will help us determine if achieving a 3dB bandwidth greater than the specified minimum value is possible. If it's not achievable, the GP will indicate that it's infeasible for the given gain, Vov, and length.

Reply:
If we ask for the optimum, I don't see how asking for a specific frequency makes sense as part of the same problem. It's like saying I want the cheapest lunch and want to know if if that's less than $5. If I have the minimum cost, it is obvious whether $5 was achievable. On the contrary, if your bound is not feasible (because you actually have no idea about the number), you get no answer. So, I see only a loss in utility from setting up the problem that way.

Reply:
I understand what you're saying. I will update the code now.